### ISSUE TITLE
Typesript import failed

### ISSUE THREAD

**Issue by Mullayam** [honojs/middleware#593]
![image](https://github.com/honojs/middleware/assets/50077856/87e2fe71-d0cd-4d53-b721-48dfd82bf401)

**Comment by Meess** [comment]
I don't think this has to do with Hono but with your bundler / packagemanger. I think your current file has the .js extension while your bundler recognises that as a CommonJS file, you should call it .mjs so the bundler recognises it as a ECMScript module which allows you to import other modules.

**Comment by Adrianmjim** [comment]
It happens to me too. It seems like an issue with the package exports.

**Comment by yusukebe** [comment]
Hi @Adrianmjim Can you share a minimal project to reproduce the problem? I will investigate it.

**Comment by yusukebe** [comment]
@Adrianmjim 

Thank you for the repo. Is it not good to add `type: "module"` to `package.json`?

```diff
diff --git a/package.json b/package.json
index 29aeaeb..97cef7d 100644
--- a/package.json
+++ b/package.json
@@ -1,5 +1,6 @@
 {
   "name": "hono-zod-validator-typescript-import-error",
+  "type": "module",
   "scripts": {
     "build": "tsc --build tsconfig.json"
   },
@@ -15,4 +16,4 @@
   },
   "version": "1.0.0",
   "packageManager": "pnpm@10.3.0"
-}
+}
\ No newline at end of file
```

**Comment by Adrianmjim** [comment]
Hi @yusukebe ,

The code should compile without using the module option. In the repository’s case, adding the module option worked due to the repository’s simplicity. I have added a slightly more complex case where adding the module option also produces an error. Even so, it should work with the commonjs option or default.

**Comment by yusukebe** [comment]
Hi @Adrianmjim 

Understood! The new version `0.4.3 ` may fix the prolemn. Can you try it?

**Comment by Adrianmjim** [comment]
Hi  @yusukebe,

Version `0.4.3` fixes the problem.

Thanks you so much!
